daemon/algod/api/server/v2/handlers.go

```diff
@@ -732,6 +732,54 @@ func (v2 *Handlers) GetBlockTxids(ctx echo.Context, round uint64) error {
 	return ctx.JSON(http.StatusOK, response)
 }
 
+func getLogsFromTxns(txns []transactions.SignedTxnWithAD, blockLogs *[]model.BlockLog, outerTxnID string) {
+
+	for _, txn := range txns {
+		byteLogs := [][]byte{}
```

Comment:
We like to make the slice the right size when we know it, so there's no growth cost.
```suggestion
		byteLogs := make([][]byte, len(txn.EvalDelta.Logs))
```

Reply:
This point can be ignored now, as I see you actually _don't_ know the eventual size, if you intend to recur for inners of inners.

Reply:
Since we are mapping the array of logs to app index and transaction ID, this array won't be increased from inners... although I suppose a single outer app could call the same app multiple times. The current behavior is to just create a new struct for each call, even if it's to an app that's already been called, but I'm open to a different approach.

Reply:
Oh, you're collecting all of a given app's logs into one struct, not one per top-level txn?  I missed that somehow.

That seems like it will remove a lot of information about what happened.

Reply:
But I also don't follow what you're saying. If you don't descend the `txn.EvalDelta.Inners[..].EvalDelta.Inners[...].EvalDelta.Inners[....]` you're never going to get all the logs.

Reply:
It does descend down through inners: https://github.com/joe-p/go-algorand/blob/6d4b23f72649f7a2fd42176ca420a424872dedcd/daemon/algod/api/server/v2/handlers.go#L749. 

I realized though I'm not actually ever getting the logs from the outers

Reply:
Ah, ok, I saw that line and somehow forgot where it was.